### PR TITLE
added prefetch check list endpoint

### PR DESCRIPTION
**Pull request by DownGoat**

### CODE REVIEW

File: src/Equinor.ProCoSys.Completion.ForeignApi/MainApi/CheckList/ICheckListApiService.cs [hunk 1]
```diff
@@ -6,7 +6,7 @@ namespace Equinor.ProCoSys.Completion.ForeignApi.MainApi.CheckList;
 
 public interface ICheckListApiService
 {
-    Task<ProCoSys4CheckList?> GetCheckListAsync(Guid checkListGuid);
+    Task<ProCoSys4CheckList?> GetCheckListAsync(Guid checkListGuid, CancellationToken cancellationToken = default);
```
Comment (janid1967) [inline src/Equinor.ProCoSys.Completion.ForeignApi/MainApi/CheckList/ICheckListApiService.cs:9]: maybe remove default and force the consumer of method to choose?

File: src/Equinor.ProCoSys.Completion.ForeignApi/MainApi/CheckList/ICheckListCache.cs [hunk 2]
```diff
@@ -1,9 +1,10 @@
 ﻿using System;
+using System.Threading;
 using System.Threading.Tasks;
 
 namespace Equinor.ProCoSys.Completion.ForeignApi.MainApi.CheckList;
 
 public interface ICheckListCache
 {
-    Task<ProCoSys4CheckList?> GetCheckListAsync(Guid checkListGuid);
+    Task<ProCoSys4CheckList?> GetCheckListAsync(Guid checkListGuid, CancellationToken cancellationToken = default);
```
Comment (janid1967) [inline src/Equinor.ProCoSys.Completion.ForeignApi/MainApi/CheckList/ICheckListCache.cs:9]: maybe remove default and force the consumer of method to choose?

File: src/Equinor.ProCoSys.Completion.WebApi/Authorizations/IAccessValidator.cs [hunk 3]
```diff
@@ -1,9 +1,10 @@
...
 using MediatR;
 
 namespace Equinor.ProCoSys.Completion.WebApi.Authorizations;
 
 public interface IAccessValidator
 {
-    Task<bool> ValidateAsync<TRequest>(TRequest request) where TRequest: IBaseRequest;
-}
\ No newline at end of file
+    Task<bool> ValidateAsync<TRequest>(TRequest request, CancellationToken cancellationToken = default) where TRequest: IBaseRequest;
```
Comment (janid1967) [inline src/Equinor.ProCoSys.Completion.WebApi/Authorizations/IAccessValidator.cs:9]: maybe remove default and force the consumer of method to choose?

File: src/Equinor.ProCoSys.Completion.WebApi/Authorizations/AccessValidator.cs
Comment (janid1967) [inline src/Equinor.ProCoSys.Completion.WebApi/Authorizations/AccessValidator.cs:1]: here U will come into conflicts with Espens branch .... maybe U merge after him, and U fix conficts?

File: src/Equinor.ProCoSys.Completion.WebApi/Authorizations/IContentAccessChecker.cs [hunk 4]
```diff
@@ -1,10 +1,12 @@
...
 
 namespace Equinor.ProCoSys.Completion.WebApi.Authorizations;
 
 public interface IContentAccessChecker
 {
-    Task<bool> HasCurrentUserAccessToCheckListAsync(Guid checkListGuid);
-    Task<bool> HasCurrentUserAccessToCheckListOwningPunchItemAsync(Guid punchItemGuid);
+    Task<bool> HasCurrentUserAccessToCheckListAsync(Guid checkListGuid, CancellationToken cancellationToken = default);
+    Task<bool> HasCurrentUserAccessToCheckListOwningPunchItemAsync(Guid punchItemGuid,
+        CancellationToken cancellationToken = default);
```
Comment (janid1967) [inline src/Equinor.ProCoSys.Completion.WebApi/Authorizations/IContentAccessChecker.cs:11]: maybe remove default and force the consumer of method to choose?

**Review by janid1967 — APPROVED** [review]

**Review by janid1967 — APPROVED** [review]
Approved after fixing tests. I guess you need to mock some external resources

File: src/Equinor.ProCoSys.Completion.WebApi/Controllers/Misc/CacheController.cs [hunk 5]
```diff
@@ -88,4 +96,21 @@ public async Task<IList<AccessablePlant>> GetPlantsFromMain()
         var plants = await _permissionApiService.GetAllPlantsForUserAsync(currentUserOid);
         return plants;
     }
+
+    [HttpGet("PrefetchCheckListByGuid/{checkListGuid}")]
+    public async Task<ActionResult> PrefetchCheckListByGuid(
```
Comment (TordJoranger) [inline src/Equinor.ProCoSys.Completion.WebApi/Controllers/Misc/CacheController.cs:101]: anonymous?

Comment (DownGoat) [reply src/Equinor.ProCoSys.Completion.WebApi/Controllers/Misc/CacheController.cs:101]: A bit unsure about that, from frontend perspective, they should ignore any errors, as they are not relevant for them for this endpoint. And it is not returning any data.

**Review by janid1967 — APPROVED** [review]

File: src/Equinor.ProCoSys.Completion.WebApi/Controllers/Misc/CacheController.cs (hunk 5 above)
Comment (TordJoranger) [reply src/Equinor.ProCoSys.Completion.WebApi/Controllers/Misc/CacheController.cs:101]: Front end does not need to care what we return, but the API should behave as expected. Next time someone uses the endpoint, maybe they do care.

File: src/Equinor.ProCoSys.Completion.WebApi/Authorizations/AccessValidator.cs
Comment (DownGoat) [reply src/Equinor.ProCoSys.Completion.WebApi/Authorizations/AccessValidator.cs:1]: Merging this and fixing the conflicts in his branch

**Review by TordJoranger — APPROVED** [review]

**Pull request merged**